000-mm2-connector-offsets-support.md

```diff
@@ -0,0 +1,390 @@
+That is because some of the fields are already known to the operator.
+Specifically, we can omit:
+* For MirrorSourceConnector:
+  * `cluster`, this is equal to `spec.mirrors[].sourceCluster` in the KafkaMirrorMaker2 resource
+  * `topic`, this is the configuration topic for the underlying Kafka Connect cluster, by default named `mirrormaker2-cluster-configs`
+* For MirrorCheckpointConnector no fields can be omitted
+* For MirrorHeartbeatConnector:
+  * `partition.sourceClusterAlias`, this is equal to `spec.mirrors[].sourceCluster` in the KafkaMirrorMaker2 resource
+  * `partition.targetClusterAlias`, this is equal to `spec.mirrors[].targetCluster` in the KafkaMirrorMaker2 resource
+
+We will also omit these from the output when listing offsets.
+This will make it easier for users to reuse the list offsets output when altering offsets.
+
+### New annotations
+
+All three actions will make use of a new annotation called `strimzi.io/connector-offsets`.
+The possible values will be `list`, `alter`, and `reset`.
+
+A second new annotation will be added called `strimzi.io/mirrormaker-connector`.
+It will be required when `strimzi.io/connector-offsets` is set.
+The value will be the name of the connector to apply the action to, for example `east-kafka->west-kafka.MirrorSourceConnector`.
+
+If the `strimzi.io/mirrormaker-connector` annotation is set, but the `strimzi.io/connector-offsets` annotation is missing the operator will take no action.
+
+### Listing offsets
+
+To list the current offsets the user will annotate the KafkaMirrorMaker2 resource with the new annotation `strimzi.io/connector-offsets` set to `list`, and the `strimzi.io/mirrormaker-connector` annotation.
+After the annotations are added, on the next reconciliation the operator will fetch the current offsets for the connector and create a Kubernetes ConfigMap containing the JSON response.
+The operator will add an owner reference to the Kubernetes ConfigMap pointing to the KafkaMirrorMaker2 resource that was annotated.
+If the ConfigMap already exists the operator will patch the ConfigMap with updated data and not add an owner reference.
+This means any existing keys in the ConfigMap that the operator is not updating will remain as before.
+The operator will then remove the annotations from the KafkaMirrorMaker2 CR.
+
+If the user wants to see an updated set of offsets they will need to re-annotate the resource.
+
+The name of the ConfigMap the operator creates/updates will be set by the user in the KafkaMirrorMaker2 CR.
+The KafkaMirrorMaker2 CRD will be updated to contain a new top-level (i.e. directly under `spec`) property called `listOffsets` that must be set for the `strimzi.io/connector-offsets=list` annotation to take effect.
+The structure of the `listOffsets` property will be as below, to allow it to be extended in future if required:
+
+```yaml
+listOffsets:
+  toConfigMap:
+    name: my-connector-offsets
```

Comment:
I think you should place it with the connector configuration.
* That is where you stop the connector
* It will IMHO better match to how the other options work
* It will also fit the API model as you can inherit it with the Connect part of the implementation.

Reply:
Good point, that makes more sense, I've updated it.